microprofile/WFLY-12228_upgrade_microprofile-health_2.0.1.adoc

```diff
@@ -0,0 +1,82 @@
+= [WFLY-12228] Upgrade MicroProfile Health 2.0.1.
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+WildFly is providing MicroProfile Health 1.0 for applications deployed in WildFly.
+
+MicroProfile 3.0 has been released with a new major version for MicroProfile Health 2.0.1 that contains new features and breaking changes as described in its https://github.com/eclipse/microprofile-health/releases/tag/2.0.1[release notes].
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12228[WFLY-12228]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1206[EAP7-1206]
+* https://issues.jboss.org/browse/CLOUD-3326[CLOUD-3326] - Integrate MicroProfile Health 2.0.1 HTTP endpoints in EAP probes
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+* [x] QE
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile Health is provided by https://github.com/smallrye/smallrye-health[smallrye-health].
+
+smallrye-health uses CDI to discover all health check probes and provides a `SmallryeHeathReporter` that is `@ApplicationScope`-d.
+
+WildFly extensions are not using CDI and the health probes must be available even when there is no deployed application.
+The microprofile-health-smallrye will provide its own "reporter" that aggreated healtch checks procecures when applications are deployed/undeployed.
+
+
+== Requirements
+
+* Support Eclipse MicroProfile Health 2.0.1 for applications deployed in WildFly.
+* Provide two additional HTTP endpoints `/health/live` and `/health/ready` to check exclusilvely liveness checks probes and readiness check probes.
+* Provide two additional management operations `:check-live` and `:check-ready` to check exclusilvely liveness checks probes and readiness check probes.
+* Repurpose the existing HTTP endpoint `/health` and `:check` operation to check *all* check probes (both liveness and readiness probes)
+* Provides two management attributes to specify the global status when no readiness or liveness probes are defined. This allows applications to report `DOWN` until their probes verify that the application is ready/live. By default, these attributes will report `UP`.
```

Comment:
I think the Galleon layer that configures the default variant of the mp health subsystem needs to make it easy for cloud users to do this using env vars. Something like 

empty-readiness-checks-status="${env.SOME_ENV_VAR_NAME_TBD:UP}"

We tend to think in terms of system props for expressions in our standard configs, but for cloud use cases env vars are more useful.

Reply:
good point, I'll update the feature PR to use an expression resolved by an env var at https://github.com/wildfly/wildfly/pull/12392/files#diff-44104d2318acaca0ae7b10645fb14056